src/ui/flowgraphitem.cpp

```diff
@@ -690,6 +697,30 @@ void FlowGraphItem::drawAddSourceDialog(FlowGraph *fg) {
                         }
                     }
 
+                    if (c.name == "Query signals") {
+                        querySignalFilters.drawFilters();
+
+                        float windowWidth = ImGui::GetWindowWidth();
+                        float buttonPosX  = windowWidth - ImGui::GetStyle().ItemSpacing.x - ImGui::GetStyle().FramePadding.x - ImGui::CalcTextSize("Add Filter").x;
+                        ImGui::SetCursorPosX(buttonPosX);
+                        if (ImGui::Button("Add Filter")) {
+                            querySignalFilters.emplace_back(QueryFilterElement{ querySignalFilters });
+                        }
+                        ImGui::Separator();
+                        ImGui::SetNextWindowSize(ImGui::GetContentRegionAvail(), ImGuiCond_Once);
+                        ImGui::BeginChild("Signals");
+                        signalList.drawElements();
+
+                        float refreshButtonPosX = ImGui::GetWindowWidth() - ImGui::GetStyle().ItemSpacing.x - ImGui::GetStyle().FramePadding.x - ImGui::CalcTextSize("Refresh").x;
+                        float refreshButtonPosY = ImGui::GetWindowHeight() - ImGui::GetStyle().ItemSpacing.y - ImGui::GetStyle().FramePadding.y - ImGui::CalcTextSize("Refresh").y;
+                        ImGui::SetCursorPos({ refreshButtonPosX, refreshButtonPosY });
+                        if (ImGui::Button("Refresh")) {
+                            signalList.update();
+                        }
+
+                        ImGui::EndChild();
```

Comment:
`ImGui::EndChild` is an exception to the rule: it has to be called regardless of the return value of `ImGui;;BeginChild` (same as bare `ImGui::Begin`).

See: https://github.com/ocornut/imgui/blob/master/imgui.h#L331-L334

There are also some other Begin statements where the return value is ignored which will fail as the corresponding End is called unconditionally.

Reply:
thanks for fixing. Performance wise leaving the if statement in place and just moving out the End() should save a bit of needless computation, but the effects are probably neglible